### PR TITLE
Feature/unify search

### PR DESCRIPTION
**Pull request by dabeshouse**
Unify is an optimistic name for the branch.
The behavior is now unified, but not the code behind it.

### CODE REVIEW

**Comment by dabeshouse** [comment]
Reviewed my own PR - trust me!
Merging in to get complete builds for testing.

File: core/ide/src/main/java/org/alice/ide/croquet/models/project/find/core/FindContentManager.java [hunk 1]
```diff
@@ -161,103 +158,64 @@ private SearchResult checkFind(Object searchObject) {
...
 
-  private List<SearchResult> sortByRelevance(String string, List<SearchResult> searchResults) {
+  private List<SearchResult> sortByRelevance(String[] terms, List<SearchResult> searchResults) {
     List<SearchResult> unsortedList = Lists.newArrayList(searchResults);
-    List<SearchResult> rv = Lists.newArrayList();
     Map<SearchResult, Double> scoreMap = Maps.newHashMap();
     for (SearchResult obj : unsortedList) {
-      scoreMap.put(obj, score(obj, string));
+      scoreMap.put(obj, score(obj, terms));
     }
```
Comment (jennej) [inline core/ide/src/main/java/org/alice/ide/croquet/models/project/find/core/FindContentManager.java:192]: 👏👏👏

File: core/ide/src/main/java/org/alice/ide/croquet/models/project/find/core/FindContentManager.java [hunk 2]
```diff
@@ -161,103 +158,64 @@ private SearchResult checkFind(Object searchObject) {
...
-    double rv = 0;
-    if (obj.getName().equals(string)) {
-      rv += 2;
-    }
-    if (obj.getName().contains(string)) {
-      rv += 1;
-    }
-    if (obj.getName().toLowerCase().startsWith(string.toLowerCase())) {
-      rv += 1;
+  // Produce a zero or negative score.
```
Comment (jennej) [inline core/ide/src/main/java/org/alice/ide/croquet/models/project/find/core/FindContentManager.java:197]: and comments! You beautiful thing.

File: core/ide/src/main/java/org/alice/stageide/gallerybrowser/search/core/SearchGalleryWorker.java [hunk 3]
```diff
@@ -52,97 +52,83 @@
...
+      for (ResourceNode child : node.getNodeChildren()) {
+        appendMatches(matches, child);
+      }
+    }
+  }
+
+  private boolean allTermsMatch(ResourceNode node) {
+    for (String term : terms) {
+      boolean termFound = false;
+      String[] tags = node.getResourceKey().getTags();
```
Comment (jennej) [inline core/ide/src/main/java/org/alice/stageide/gallerybrowser/search/core/SearchGalleryWorker.java:108]: tags can be final? is my C++ showing too hard?

**Review by jennej — COMMENTED** [review]
sorry i'm slow, but you're doing good things!

**Pull request merged**